acceptance/tests/allow_arbitrary_node_name_fact_for_agent.rb

```diff
@@ -11,9 +11,109 @@
 
 node_names.uniq!
 
-authfile = "#{testdir}/auth.conf"
-authconf = node_names.map do |node_name|
-  %Q[
+if @options[:is_puppetserver]
+  step "Ensure legacy auth.conf is disabled for test" do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => false}})
+  end
+
+  teardown do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
```

Comment:
Right now the puppetserver setup forcibly sets `use-legacy-auth-conf` to true so the original value can't vary. Once we have fixes to the rest of the tests I want to drop these setup steps entirely and assume that if we're running under puppetserver then we're using the tk-auth rules.

I'm fine with backing up the entire puppetserver configuration directory though.